sbi/utils/sbiutils.py

```diff
@@ -948,3 +952,181 @@ def mog_log_prob(
     exponent = -0.5 * utils.batched_mixture_vmv(precisions_pp, theta_minus_mean)
 
     return torch.logsumexp(weights + constant + log_det + exponent, dim=-1)
+
+
+def get_log_abs_det_jacobian(
+    transform,
+    parameters_constrained: Tensor,
+    parameters_unconstrained: Tensor,
+) -> Tensor:
+    """Return log_abs_det_jacobian over batch of parameters.
+
+    Take sum over second dimension if needed.
+
+    Args:
+        transform: transform applied to the parameters
+        parameters_constrained: parameters in untransformed space
+        parameters_unconstrained: parameters in transformed space
+
+    Returns:
+        vals: log_abs_det_jacobian value for each batch entry.
+    """
+
+    batch_size, dim_parameters = parameters_constrained.shape
+
+    vals = transform.log_abs_det_jacobian(
+        parameters_constrained, parameters_unconstrained
+    )
+
+    # For identity transforms log_abs_det_jacobian is returned for each dimension.
```

Comment:
I don't get this. Any `torch.transform` seems to behave like that:
```python
import torch.distributions.transforms as torch_tf

torch_sigmoid = torch_tf.SigmoidTransform()

a = torch.tensor([[3., 2.0], [-2.5, 2.1]])
b = torch_sigmoid(a)
print(torch_sigmoid.log_abs_det_jacobian(a, b)) # shape 2x2
```

Reply:
ok, maybe it was changed again, I will have a look..

Reply:
indeed, this is not needed anymore, I will remove the helper function.

Reply:
actually, I checked again and found that in some cases the call to `log_abs_det_jacobian` is returned already summed over the second dimension, e.g., when the transform is an `IndependentTransform` which seems to automatically set `reinterpreted_batch_dims`, see [here for documentation](https://pytorch.org/docs/stable/_modules/torch/distributions/transforms.html#IndependentTransform):
```python
prior = Boxuniform(zeros(2), ones(2))
transform = biject_to(prior.support).inv
print(transform)

>>> _InverseTransform(IndependentTransform(ComposeTransform(
    SigmoidTransform(),
    AffineTransform()
), 1))

theta = prior.sample((10,))
transformed_theta = transform(theta)
print(transform.log_abs_det_jacobian(theta, transformed_theta).shape)

>>> torch.Size([10])
```

Reply:
I think this is because `BoxUniform` is an `Independent` distribution.